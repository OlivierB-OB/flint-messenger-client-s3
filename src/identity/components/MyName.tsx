import Typography from '@material-ui/core/Typography';
import React from 'react';
import { connect } from 'react-redux';
import { IAppState } from '../../appReducer';
import { UserName, IUserNameProps } from '../../users/components/UserName';

type IMyNameDisplayProps = Partial<IUserNameProps>;

function MyNameDisplay({ info }: IMyNameDisplayProps) {
  if (!info) return null;
  return (
    <Typography variant="h6">
      <UserName info={info} />
    </Typography>
  );
}

const mapStateToProps = ({ identity }: IAppState) => ({
  info: identity.info,
});

export const MyName = connect(mapStateToProps)(MyNameDisplay);
