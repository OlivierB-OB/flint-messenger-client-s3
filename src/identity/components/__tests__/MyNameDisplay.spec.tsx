import React from 'react';
import { shallow } from 'enzyme';
import { MyNameDisplay } from '../MyName';

describe('MyNameDisplay', () => {
  it('should return null if info is undefined', async () => {
    const props: any = {};
    const component = shallow(<MyNameDisplay {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should display the username given the provided info', async () => {
    const props: any = {
      info: 'the user information',
    };
    const component = shallow(<MyNameDisplay {...props} />);
    expect(component).toMatchSnapshot();
  });
});
