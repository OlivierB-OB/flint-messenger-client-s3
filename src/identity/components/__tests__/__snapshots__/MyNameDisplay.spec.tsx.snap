// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyNameDisplay should display the username given the provided info 1`] = `
<WithStyles(ForwardRef(Typography))
  variant="h6"
>
  <UserName
    info="the user information"
  />
</WithStyles(ForwardRef(Typography))>
`;

exports[`MyNameDisplay should return null if info is undefined 1`] = `""`;
